src/YoutubeAtom.tsx

```diff
@@ -64,8 +66,14 @@ type YoutubeAtomType = {
     title?: string;
     duration?: number; // in seconds
     origin?: string;
+    gaEventEmitter?: (event: VideoEventKey) => void;
+    ophanEventEmitter?: (event: VideoEventKey) => void;
 };
 
+type YoutubeStateChangeEvent = { data: -1 | 0 | 1 | 2 | 3 | 5 };
+
+let progressTracker: NodeJS.Timeout | null;
```

Comment:
I _think_ this is scoped but it'd be interesting to have some test cases where there are multiple youtube elements on the same page

Reply:
This is a really good point. I quickly tested it out in a dummy component, looks like it is scoped to that component's rendering so it should be find. But thanks for raising this. If you want to try it out here is the code I used:

`SomeTest.tsx`
```
import React from 'react';

let someRandomNumber;

export const SomeTest = () => {
    someRandomNumber = Math.random();
    return <div>{someRandomNumber}</div>;
};
```

`SomeTest.stories.tsx`
```
import React from 'react';

import { SomeTest } from './SomeTest';

export default {
    title: 'SomeTest',
};

export const SomeTestDefault = () => (
    <>
        <SomeTest />
        <SomeTest />
        <SomeTest />
    </>
);
```

Output:
<img width="258" alt="Screenshot 2020-11-18 at 11 29 49" src="https://user-images.githubusercontent.com/8831403/99525197-6a93cd00-2991-11eb-931c-75f92208fe9a.png">

Reply:
Related, but different issue: It might be good to comment here that the reason this setInterval _isn't_ in a `useEffect` is that it's not setting state. If someone were to come along at a later stage and drop a setMyStateValue hook function call here it [won't work](https://stackoverflow.com/questions/53024496/state-not-updating-when-using-react-state-hook-within-setinterval).

Explanation: https://overreacted.io/making-setinterval-declarative-with-react-hooks/

On the other hand, maybe we can put it in one? We're hydrating things anyway so... But there is a cost to this in terms of the code not running until after the render so we should think this through but as far as I can tell this is not time sensitive code and delaying to post render could even be a perf win